frontend/src/components/PreviewModal.tsx

```
@@ -31,14 +31,25 @@ const PreviewModal: React.FC<PreviewModalProps> = ({
-            {messages.map(message => (
-              <div key={message.id} className="p-4 bg-gray-100 rounded-lg">
-                <p className="text-gray-800">{message.content}</p>
-                <p className="text-sm text-gray-500 mt-2">
-                  {new Date(message.timestamp).toLocaleString()}
-                </p>
-              </div>
-            ))}
+            {messages.map((message) => {
+              // Support both `content` and legacy `text` fields
+              const content =
+                // @ts-ignore – allow access to `text` for fallback support
```

ESLint

[error] Use "@ts-expect-error" instead of "@ts-ignore", as "@ts-ignore" will do nothing if the following line is error-free.

(@typescript-eslint/ban-ts-comment)